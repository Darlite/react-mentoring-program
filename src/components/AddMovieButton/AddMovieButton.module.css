.addMovieButton {
    color: #F65261;
    text-transform: uppercase;
    font-weight: 600;
    font-size: 18px;
    padding: 15px 20px;
    background-color: rgba(255, 255, 255, 0.2);
    border-radius: 4px;
    position: absolute;
    top: 15px;
    right: 380px;
    cursor: pointer;
}
