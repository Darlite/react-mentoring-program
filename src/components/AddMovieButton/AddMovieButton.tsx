import React from "react";
import styles from "../AddMovieButton/AddMovieButton.module.css";
import {DialogType} from "../../constants/DialogType";

interface AddMovieButtonProps {
    handleShowDialog: (dialogType: DialogType) => void;
}

const AddMovieButton: React.FC<AddMovieButtonProps> = ({ handleShowDialog }) => {
    return (
        <span className={styles.addMovieButton}
              id="addMovieButton"
              role="button"
              onClick={() => handleShowDialog(DialogType.AddMovie)}
        >
            + Add movie
        </span>
    )
}

export default AddMovieButton;