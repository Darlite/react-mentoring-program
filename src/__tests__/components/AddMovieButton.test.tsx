import React from "react";
import {render, screen} from "@testing-library/react";
import AddMovieButton from "../../components/AddMovieButton/AddMovieButton";
import userEvent from "@testing-library/user-event";

describe("AddMovieButton", () => {
    it("renders correctly", () => {
        const { container } = render(<AddMovieButton handleShowDialog={jest.fn}/>);
        expect(container).toMatchSnapshot();
    });

    it("handles a click correctly,", () => {
        const handleShowDialog = jest.fn();
        render(<AddMovieButton handleShowDialog={handleShowDialog}/>);

        const addButton = screen.getByRole("button");

        userEvent.click(addButton);
        expect(handleShowDialog).toHaveBeenCalled();
    });
})