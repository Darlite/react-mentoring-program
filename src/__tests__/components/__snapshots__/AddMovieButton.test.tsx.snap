// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieButton renders correctly 1`] = `
<div>
  <span
    class="addMovieButton"
    id="addMovieButton"
    role="button"
  >
    + Add movie
  </span>
</div>
`;
